refract-spec.md

```diff
@@ -105,11 +101,193 @@ In addition to expressing Refract Elements as objects with `element`, `attribute
   - (array)
   - (object)
   - (Compact Element)
-  - array[Compact Element]
+  - (array[Compact Element])
+
+##### Example
+
+Below is a Refract element `foo` that is expressed in the normal Refract representation.
+
+```json
+{
+  "element": "foo",
+  "content": "bar"
+}
+```
+
+This is how it would represented in the compact version.
+
+```json
+["foo", {}, {}, "bar"]
+```
+
+## Primitive Elements
+
+Primitive Elements extend upon the base Element to define elements based on the Primitive Types of Refract.
+
+A Refract document MUST support each of these primitive elements. These elements MAY be extended or replaced based on namespaces provided in any root or parent element.
+
+### Null Element (Element)
+
+A `Null Element` is an element that has a Null Type as its value. It extends upon the Refract Element.
+
+#### Properties
+
+- `element`: null (string, fixed)
+- `content` (enum)
+    - (null)
+    - (Null Element)
 
 #### Example
 
-This example will show a JSON object represented as both a normal Refract Element and a Compact Element.
+In JSON, `null` is represented as `null`.
+
+```json
+null
+```
+
+In Refract, it is represented as a `Null Element` element.
+
+```json
+{
+  "element": "null",
+  "content": null
+}
+```
+
+### String Element (Element)
+
+A String Element provides an element for Refract String Types.
+
+#### Properties
+
+- `element`: string (string, fixed)
+- `content` (enum)
+    - (string)
+    - (String Element)
```

Comment:
:+1: for taking this out. I think it is not needed right now and we can always add this later. The simpler the initial version is going to be the better.